arangod/Pregel/Worker/Handler.h

```diff
@@ -296,13 +326,45 @@ struct WorkerHandler : actor::HandlerBase<Runtime, WorkerState<V, E, M>> {
     prepareGlobalSuperStep(std::move(message));
 
     // resend all queued messages that were dedicatd to this gss
-    for (auto const& message : this->state->messagesForNextGss) {
+    for (auto const& messageNextGss : this->state->messagesForNextGss) {
       this->template dispatch<worker::message::PregelMessage>(this->self,
-                                                              message);
+                                                              messageNextGss);
     }
     this->state->messagesForNextGss.clear();
 
-    auto verticesProcessed = processVertices();
+    std::vector<futures::Future<ResultT<VerticesProcessed>>> futures;
+    // first build up all future requests
+    /*
+    for (auto [idx, quiver] : enumerate(this->state->quivers)) {
+      futures.emplace_back(futures::makeFuture(
+          [idx, quiver]() { return processVertices(idx, quiver); }));
+    }
+*/
```

Comment:
```suggestion
    // first build up all future requests
```

Reply:
Wait - will not commit this. I'll clean this properly